src/ray/raylet/scheduling_resources.cc

```diff
@@ -170,16 +261,34 @@ ResourceIds::ResourceIds(double resource_quantity) {
   for (int64_t i = 0; i < whole_quantity; ++i) {
     whole_ids_.push_back(i);
   }
+  greatest_id_ = resource_quantity;
+  total_capacity_ = TotalQuantity();
+  decrement_backlog_ = 0;
 }
 
-ResourceIds::ResourceIds(const std::vector<int64_t> &whole_ids) : whole_ids_(whole_ids) {}
+ResourceIds::ResourceIds(const std::vector<int64_t> &whole_ids)
+    : whole_ids_(whole_ids),
+      greatest_id_(*std::max_element(whole_ids.begin(), whole_ids.end())),
+      total_capacity_(whole_ids.size()),
+      decrement_backlog_(0){}
 
-ResourceIds::ResourceIds(const std::vector<std::pair<int64_t, double>> &fractional_ids)
-    : fractional_ids_(fractional_ids) {}
+ResourceIds::ResourceIds(const std::vector<std::pair<int64_t, double>> &fractional_ids){
+  fractional_ids_ = fractional_ids;
+  // Find the max id and set it to greatest_id_;
+  const auto p = std::minmax_element(fractional_ids_.begin(), fractional_ids_.end());
```

Comment:
How is `max` defined for pairs?

Reply:
We can use `std::max_element` but will have to define a custom comparator to compare pairs. `std::minmax_element` has a default comparator for pairs which evaluates the second value of the pair to find the max and min.